packages/@sanity/core/src/actions/graphql/deployApiAction.js

```diff
@@ -115,8 +120,33 @@ module.exports = async function deployApiActions(args, context) {
     throw validationError ? new Error(validationError) : err
   }
 
-  if (!(await confirmValidationResult(valid, {spinner, output, prompt, force}))) {
-    return
+  // when the result is not valid and there are breaking changes afoot!
+  if (!isResultValid(valid, {spinner, force})) {
+    renderBreakingChanges(valid, {spinner, output, prompt, force, dryRun})
+
+    // not valid and a dry run? then it can exit with a error
+    if (dryRun) {
+      process.exit(1)
+    } else {
+      if (!isInteractive) {
+        throw new Error(
+          'Dangerous changes found - falling back. Re-run the command with the `--force` flag to force deployment.'
+        )
+      }
+
+      const shouldDeploy = await prompt.single({
+        type: 'confirm',
+        message: 'Do you want to deploy a new API despite the dangerous changes?',
+        default: false,
+      })
+
+      if (!shouldDeploy) {
+        return
+      }
+    }
```

Comment:
The `else` branch is not necessary, since process.exit() will immediately kill the process - kind of like an early return. Less indentation/branching makes it harder to read the code, in my opinion :)

```suggestion
    if (dryRun) {
      process.exit(1)
    }

    if (!isInteractive) {
      throw new Error(
        'Dangerous changes found - falling back. Re-run the command with the `--force` flag to force deployment.'
      )
    }

    const shouldDeploy = await prompt.single({
      type: 'confirm',
      message: 'Do you want to deploy a new API despite the dangerous changes?',
      default: false,
    })

    if (!shouldDeploy) {
      return
    }
```

Reply:
Fully agreed, committed the change :)